pkg/virtualgarden/kube_api_server_secrets.go

```diff
@@ -326,47 +326,47 @@ func (o *operation) generateNewEncryptionConfig() ([]byte, error) {
 }
 
 func (o *operation) deployKubeApiServerSecretServiceAccountKey(ctx context.Context, checksums map[string]string) error {
-	certConfig := &secretsutil.CertificateSecretConfig{
-		Name:       KubeApiServerSecretNameServiceAccountKey,
-		CertType:   secretsutil.CACert,
-		CommonName: Prefix + ":ca:kube-apiserver",
-		DNSNames: []string{
-			"virtual-garden:ca:kube-apiserver",
-		},
-	}
+	var serviceAccountKey []byte
 
-	var value []byte
+	if o.imports.VirtualGarden.KubeAPIServer.ServiceAccountKeyPem == nil ||
+		len(*o.imports.VirtualGarden.KubeAPIServer.ServiceAccountKeyPem) == 0 {
 
-	secret := o.emptySecret(KubeApiServerSecretNameServiceAccountKey)
-	err := o.client.Get(ctx, client.ObjectKeyFromObject(secret), secret)
-	if err != nil {
-		if client.IgnoreNotFound(err) != nil {
-			return err
```

Comment:
Was that a miscunderstanding?
The cluster should still be checked if no import configuration is provided!
It should only generate a new one if no one was provided + not found (like before)

Reply:
> We will introduce a new optional parameter for the key. If the import parameter is provided, we use it; otherwise we generate a new key.

Oh sorry, I have not read carefully :/

Reply:
I have changed the coding such that a new one is only generated if no one was provided + not found